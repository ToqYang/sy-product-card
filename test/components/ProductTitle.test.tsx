import React from 'react';
import renderer from 'react-test-renderer';
import { product1 } from '../data/products';
import { ProductCard, ProductTitle } from '../../src/components/';

describe('ProductTitle', () => {
  test('Must show the title right', () => {
    const wrapper = renderer.create(<ProductTitle title="Custom Product" />);
    console.log(wrapper.toJSON());

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('Send title to product Card ', () => {
    const wrapper = renderer.create(
      <ProductCard product={product1}>{() => <ProductTitle />}</ProductCard>
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
