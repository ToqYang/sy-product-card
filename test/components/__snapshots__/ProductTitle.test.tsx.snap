// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTitle Must show the title right 1`] = `
<span
  className="productDescription undefined"
>
  Custom Product
</span>
`;

exports[`ProductTitle Send title to product Card  1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription undefined"
  >
    Coffee Mug - Card
  </span>
</div>
`;
